docs/usage/sagas/efcore.md

```diff
@@ -137,4 +137,48 @@ cfg.ServiceInstance(options, instance =>
 The [Job Consumers](https://github.com/MassTransit/Sample-JobConsumers) sample is a working version of this configuration style.
 
 
+#### Multiple DbContext
+
+Multiple `DbContext` can be registered in the container which can then be used to configure sagas that are mapped by the `DbContext` and injected into other components. Calling the ```AddDbContext``` extension method will register a scoped ```DbContext``` by default. For simple scenarios where there is a single ```DbContext``` this will work. However, in scenarios where there is at least one other ```DbContext``` the dotnet command that generates Entity Framework migrations will not work as shown below.
+
+```cs
+services.AddDbContext<JobServiceSagaDbContext>(builder =>
+    builder.UseNpgsql(Configuration.GetConnectionString("JobService"), m =>
+    {
+        m.MigrationsAssembly(Assembly.GetExecutingAssembly().GetName().Name);
+        m.MigrationsHistoryTable($"__{nameof(JobServiceSagaDbContext)}");
+    }));
+
+services.AddDbContext<OrdersDbContext>(builder =>
+    builder.UseNpgsql(host.Configuration.GetConnectionString("OrdersConnection")));
+```
+
+To resolve this issue, you'll need to perform the following steps:
+
+1. Make sure that all ```DbContext``` has a constructor that takes ```DbContextOptions<TOptions>``` instead of ```DbContextOptions```.
+2. Register each ```DbContext``` with ```ServiceLifetime.Singleton``` as shown below.
```

Comment:
What's the significance of Singleton here?

Reply:
you can’t use singleton, I believe

Reply:
I was thinking Singleton here could cause some issues for sure - so questioning whether/why it's required here.

Reply:
Probably should elaborate a little here. If you have a component, ComponentA, that is registered as a singleton and a non-MassTransit DbContext is injected as scoped (i.e. default call to AddDbContext) then the EF command will throw the following exception:

An error occurred while accessing the Microsoft.Extensions.Hosting services. Continuing without the application service provider. Error: Some services are not able to be constructed (Error while validating the service descriptor 'ServiceType: Services.Core.IKitchenManager Lifetime: Singleton ImplementationType: Services.Core.KitchenManager': Cannot consume scoped service 'Data.Core.OrdersDbContext' from singleton 'Services.Core.IKitchenManager'.) (Error while validating the service descriptor 'ServiceType: Services.Core.IPrepareOrder Lifetime: Singleton ImplementationType: Services.Core.PrepareOrder': Cannot consume scoped service 'Data.Core.OrdersDbContext' from singleton 'Services.Core.IPrepareOrder'.) (Error while validating the service descriptor 'ServiceType: Services.Core.ILowInventoryDetector Lifetime: Singleton ImplementationType: Services.Core.LowInventoryDetector': Cannot consume scoped service 'Data.Core.OrdersDbContext' from singleton 'Services.Core.ILowInventoryDetector'.) (Error while validating the service descriptor 'ServiceType: RestaurantService.Core.Consumers.PrepareOrderItemConsumer Lifetime: Scoped ImplementationType: RestaurantService.Core.Consumers.PrepareOrderItemConsumer': Cannot consume scoped service 'Data.Core.OrdersDbContext' from singleton 'Services.Core.IPrepareOrder'.) (Error while validating the service descriptor 'ServiceType: Quartz.Spi.IJobFactory Lifetime: Singleton ImplementationType: RestaurantService.Core.RestaurantJobFactory': Cannot consume scoped service 'Data.Core.OrdersDbContext' from singleton 'Services.Core.ILowInventoryDetector'.) (Error while validating the service descriptor 'ServiceType: RestaurantService.Core.LowInventoryDetectorJob Lifetime: Transient ImplementationType: RestaurantService.Core.LowInventoryDetectorJob': Cannot consume scoped service 'Data.Core.OrdersDbContext' from singleton 'Services.Core.ILowInventoryDetector'.)
Unable to create an object of type 'RestaurantServiceDbContext'. For the different patterns supported at design time, see https://go.microsoft.com/fwlink/?linkid=851728

The way to fix this so that the EF command works is to register the DbContext as singleton. I need to make a change to the docs to specify that the Saga DbContext does not need to be changed. This step would be optional if you are injecting a DbContext into a component registered as a singleton.

Reply:
Ah ha - yes this looks very specific to the setup having IKitchenManager as a singleton. I suspect the Saga DB Context will be fine

Reply:
I think step 2 is optional

Reply:
it is not even optional, you shouldn't propagate incorrect db usage. DB context should be either scoped or transient. Any time you don't need context you should dispose it, also you need to clear change tracker to make it work correctly

Reply:
there is, some databases where client designed to handle connections, but EF Core is not the case, it is supposed to be disposed every time you done with your flow

Reply:
I have to make some changes to how Quartz was being registered. The Quartz DI allows you to call a method for registering the custom jobs, the caveat being it registers the job factory as a singleton. To get around this I needed to call the UseMicrosoftDependencyInjectionScopedJobFactory instead and now I can register my dependencies as scoped. Step 2 can be removed. EF Core wasn't the problem, it was the fact that I was calling the wrong job factory registration method in Quartz DI that made me go down the path of registering those components as singleton.